tests/ios/xcodeproj/Test-BuildForDevice-Project.xcodeproj/project.pbxproj

```diff
@@ -409,7 +409,7 @@
 				CLANG_ENABLE_OBJC_ARC = YES;
 				FRAMEWORK_SEARCH_PATHS = "$(PLATFORM_DIR)/Developer/Library/Frameworks";
 				GCC_PREPROCESSOR_DEFINITIONS = "$(inherited)";
-				HEADER_SEARCH_PATHS = "\"$BAZEL_WORKSPACE_ROOT/bazel-out/ios-arm64-min10.0-applebin_ios-ios_arm64-dbg-ST-c32106249197/bin/tests/ios/app/OnlySources_public_hmap.hmap\" \"$BAZEL_WORKSPACE_ROOT/bazel-out/ios-arm64-min10.0-applebin_ios-ios_arm64-dbg-ST-c32106249197/bin/tests/ios/app/OnlySources_private_hmap.hmap\" \"$BAZEL_WORKSPACE_ROOT\"";
+				HEADER_SEARCH_PATHS = "\"$BAZEL_WORKSPACE_ROOT/bazel-out/ios-x86_64-min10.0-applebin_ios-ios_x86_64-dbg-ST-7bf874b56ea0/bin/tests/ios/app/OnlySources_public_hmap.hmap\" \"$BAZEL_WORKSPACE_ROOT/bazel-out/ios-x86_64-min10.0-applebin_ios-ios_x86_64-dbg-ST-7bf874b56ea0/bin/tests/ios/app/OnlySources_private_hmap.hmap\" \"$BAZEL_WORKSPACE_ROOT\"";
```

Comment:
`x86_64` is definitely not expected here when generating a proj that can build for device. I'd expect this to continue to be `arm64`

Reply:
I just ran what was recommended by https://github.com/bazel-ios/rules_ios/pull/298#pullrequestreview-743277013

Reply:
Just encountered this in another PR (#302). To match the existing projects, it looks like one needs to run:
```
# Regenerate all projects
bazelisk query 'kind(xcodeproj, ...)' | xargs -n 1 bazelisk run
# Regenerate the project that builds for device _only_ targeting arm64
bazelisk run //tests/ios/xcodeproj:Test-BuildForDevice-Project --ios_multi_cpus=arm64
```
(Figured this out by reading the script CI is running: https://github.com/bazel-ios/rules_ios/blob/f5515c35135d7016f12cf3c960af6d01ba9e8ef1/tests/xcodeproj-tests.sh#L58)

Reply:
@nkoroste can you try that and see if it fixes the expected archs here? We can then iterate on the remaining CI failure

@gyfelton did you end up figuring out what was causing the other failures? Worst case if not related to this PR we could land it once the archs issue is fixed

Reply:
> @nkoroste can you try that and see if it fixes the expected archs here? We can then iterate on the remaining CI failure

how?

Reply:
See comment above: https://github.com/bazel-ios/rules_ios/pull/298#discussion_r709350065

Reply:
done